go-controller/pkg/ovn/gateway_cleanup.go

```diff
@@ -97,9 +93,59 @@ func gatewayCleanup(nodeName string, nodeSubnets []*net.IPNet) error {
 			}
 		}
 	}
+
+	// We don't know the gateway mode as this is running in the master, try to delete the additional local
+	// gateway for the shared gateway mode. it will be no op if this is done for other gateway modes.
+	delPbrAndNatRules(nodeName)
 	return nil
 }
 
+func delPbrAndNatRules(nodeName string) {
+	// delete the dnat_and_snat rule that we added for the management port IP
+	// Note: we don't need to delete any MAC bindings that are dynamically learned from OVN SB DB
+	// because there will be none since this NAT is only for outbound traffic and not for inbound
+	mgmtPortName := "k8s-" + nodeName
+	externalIP, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=external_ip", "find", "nat", fmt.Sprintf("logical_port=%s", mgmtPortName))
+	if err != nil {
+		klog.Errorf("Failed to fetch the dnat_and_snat rule for the management port %s "+
+			"stderr: %s, error: %v", mgmtPortName, stderr, err)
+		externalIP = ""
+	}
+	if externalIP != "" {
+		_, stderr, err = util.RunOVNNbctl("--if-exists", "lr-nat-del", ovnClusterRouter, "dnat_and_snat", externalIP)
+		if err != nil {
+			klog.Errorf("Failed to delete the dnat_and_snat ip %s associated with the management "+
+				"port %s: stderr: %s, error: %v", externalIP, mgmtPortName, stderr, err)
+		}
+	}
+
+	// find the pbr rules associated with the node and delete them
+	matches, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading", "--columns=match",
+		"find", "logical_router_policy")
+	if err != nil {
+		klog.Errorf("Failed to fetch the policy route for the node subnet "+
+			"stdout: %s, stderr: %s, error: %v", matches, stderr, err)
+		matches = ""
+	}
+	nodeSubnetMatchSubStr := fmt.Sprintf("rtos-%s", nodeName)
+	for _, match := range strings.Split(matches, "\n\n") {
+		var priority string
+		if strings.Contains(match, nodeSubnetMatchSubStr) {
+			priority = nodeSubnetPolicyPriority
+		} else if strings.Contains(match, nodeName) {
+			priority = mgmtPortPolicyPriority
+		} else {
+			continue
```

Comment:
that shouldn't happen because we are the ones adding the rules and the only rules allowed are `nodeSubnetPolicyPriority` and `mgmtPortPolicyPriority`, but i will log it.

Reply:
Also, I am walking through all the PBR rules in the cluster to clear up the PBR rules for the node that is being removed from the cluster. So, I will find rules that don't match the current node, so I cannot  print them in the `else` branch above.